tests/scanner/scanners/external_project_access_scanner_test.py

```diff
@@ -0,0 +1,301 @@
+
+
+class GetUserEmailsTest(ForsetiTestCase):
+    """Test the storage_helpers modue."""
+    def setUp(self):
+        self.engine = create_test_engine()
+        _session_maker = sessionmaker()
+        self.session = _session_maker(bind=self.engine)
+        initialize(self.engine)
+        res_user1 = ResourceMock('1', {'primaryEmail': 'user1@example.com',
+                                       'suspended': False},
+                                 'gsuite_user',
+                                 'resource')
+        res_user2 = ResourceMock('2', {'primaryEmail': 'user2@example.com',
+                                       'suspended': False},
+                                 'gsuite_user',
+                                 'resource')
+        res_user3 = ResourceMock('3', {'primaryEmail': 'user3@example.com',
+                                       'suspended': False},
+                                 'gsuite_user',
+                                 'resource')
+        self.resources = [res_user1, res_user2, res_user3]
+        self.storage = Storage(self.session)
+        _ = self.storage.open()
+        for resource in self.resources:
+            self.storage.write(resource)
+        self.storage.commit()
+        self.service_config = TestServiceConfig()
+
+    #pylint: disable=C0301,W9016,W9015,W0613
+    @mock.patch('google.cloud.forseti.scanner.scanners.external_project_access_scanner._get_inventory_storage')
+    @mock.patch('google.cloud.forseti.services.inventory.storage.DataAccess.get_latest_inventory_index_id')
+    def test_get_emails(self, mock_get_latest_inv_ndx_id, mock_storage):
+        """Test retrieving e-mails from storage"""
+
+        expected_emails = [u'user1@example.com',
+                           u'user2@example.com',
+                           u'user3@example.com']
+        mock_storage.return_value = self.storage
+        emails = epas.get_user_emails(self.service_config)
+        self.assertListEqual(emails, expected_emails)
+
+
+class OtherModuleTest(ForsetiTestCase):
+    """Test misc."""
+
+    def setUp(self):
+        """Set up."""
+        fake_global_configs = {
+            'crm': {'max_calls': 4, 'period': 1.2}}
+
+        mock.patch.object(google.auth, 'default', return_value=(
+            mock.Mock(spec_set=credentials.Credentials), 'test-project'))
+
+        mock_creds = api_helpers.get_delegated_credential('user1@eaxmple.com')
+
+        self.crm_api_client = crm.CloudResourceManagerClient(
+            global_configs=fake_global_configs,
+            use_rate_limiter=False,
+            credentials=mock_creds)
+        self.project_id = fake_crm_responses.FAKE_PROJECT_ID
+
+    def test_get_project_ids(self):
+        """Test get_projects_ids() for all lifecycle_states."""
+        response = json.dumps(fake_crm_responses.FAKE_PROJECTS_API_RESPONSE1)
+        http_mocks.mock_http_response(response)
+
+        proj_ids = epas.extract_project_ids(self.crm_api_client)
```

Comment:
Let's stick with the full spelling here `project_ids`.

Reply:
modified as suggested